pkg/build/controller/controller.go

```diff
@@ -102,9 +102,13 @@ func (bc *BuildController) HandleBuild(build *buildapi.Build) error {
 	}
 
 	if buildutil.IsBuildComplete(build) {
+		if _, ok := build.Annotations[buildapi.BuildCompletedAnnotation]; ok {
+			return nil
+		}
```

Comment:
> @smarterclayton how is building a cache better ? It really doesn't matter that we run the policy.OnComplete again, but we'd rather avoid it if we can.

Adding post-processing annotations instead of just using the straight status fields we have (you can index on them if you want) is effectively adding multiple (potentially conflicting) sources of truth.  Adding them to annotations in particular gets weird quick, since you've created a "status" annotation you could set via a spec update.  In addition, you don't even get label selection benefit as an annotation and you aren't using them for restricting selection from the server, so they aren't buying much in terms of performance over checking the actual status.